src/ir/context.rs

```diff
@@ -319,7 +319,8 @@ impl<'ctx> BindgenContext<'ctx> {
                item,
                definition);
 
-        assert!(item.is_named());
+        assert!(*item.as_type().unwrap().kind() == TypeKind::Named,
```

Comment:
nit: `item.expect_type().kind().is_named()`?

If the `PartialEq` commit is due to this assert, I'd rather remove it and add a convenience method to `TypeKind`, or use the `matches` crate.